### PR TITLE
fix: hanging on basic auth requests in chromium browsers

### LINKED ISSUE #27789: Basic auth in baseUrl not working since v13
### Current behavior

Passing Basic Auth in baseUrl inside of `cypress.config.ts` 
`baseUrl: 'https://username:password@google.com'`  does not load specs or other menus in the runner.

`baseUrl: 'https://google.com'` Cypress correctly loads the specs when opened

### Desired behavior

Passing Basic Auth in baseUrl inside of `cypress.config.ts` before v13 correctly loads and runs specs.

### Test code to reproduce

Any random test using the mentioned base URLs above, the issue happens after `npx cypress open` not in a test code itself

### Cypress Version

13.1.0

### Node version

v18.16.1

### Operating System

macOS

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by ryanthemanuel**
### Additional details

Fixes a regression in `13.0.0` where tests using Basic Authorization can sometimes hang indefinitely on chromium browsers. This is caused because our assets are loaded twice when we have basic auth set on the base url. It is unclear why this is happening, but this has been happening since at least 12.0.0 it just didn't cause a problem until we started relying on a top level local variable for a cache of CDP sockets. The fix for this is to set the cache on window instead. This is not ideal, and we should investigate why this behavior is happening, but it is worth getting a fix out there for now so that we don't hang in this scenario.

### Steps to test

The new system test tests this scenario.

### How has the user experience changed?

Tests will no longer hang

### PR Tasks

- [X] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/socket/lib/browser.ts [hunk 1]
```diff
@@ -4,18 +4,26 @@ import type { SocketShape } from './types'
...
 export type { Socket } from 'socket.io-client'
 
-const sockets: {[key: string]: CDPBrowserSocket} = {}
+declare global {
+  interface Window {
+    cypressSockets: {[key: string]: CDPBrowserSocket}
+  }
+}
+
+window.cypressSockets ||= {}
```
Comment (mjhenkes) [inline packages/socket/lib/browser.ts]: why do you need to add sockets to window for this fix? Is it for testing?

File: packages/socket/lib/browser.ts [hunk 2]
```diff
@@ -4,18 +4,26 @@ import type { SocketShape } from './types'
...
     const fullNamespace = `${opts?.path}${uri}`
 
-    if (!sockets[fullNamespace]) {
-      sockets[fullNamespace] = new CDPBrowserSocket(fullNamespace)
+    if (!window.cypressSockets[fullNamespace]) {
+      window.cypressSockets[fullNamespace] = new CDPBrowserSocket(fullNamespace)
     }
 
-    return sockets[fullNamespace]
+    window.cypressSockets[fullNamespace].connect()
```
Comment (mjhenkes) [inline packages/socket/lib/browser.ts:27]: What does connect do that we weren't doing before and why does it fix the hang?

**Review by chrisbreiding — APPROVED** [review]

File: packages/socket/lib/browser.ts (hunk 2 above)
Comment (ryanthemanuel) [reply packages/socket/lib/browser.ts:27]: Added some doc to explain this a little better: [`27b44ed` (#27781)](https://github.com/cypress-io/cypress/pull/27781/commits/27b44ed1a67b17e529b5e8016c8470587cc9f239)

File: packages/socket/lib/browser.ts (hunk 1 above)
Comment (ryanthemanuel) [reply packages/socket/lib/browser.ts]: Added some doc to explain this a little better: [`27b44ed` (#27781)](https://github.com/cypress-io/cypress/pull/27781/commits/27b44ed1a67b17e529b5e8016c8470587cc9f239)

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**